services/shotgrid_common/ayon_shotgrid_hub/update_from_shotgrid.py

```diff
@@ -242,7 +249,27 @@ def update_ayon_entity_from_sg_event(
     )
 
     if not sg_ay_dict["data"].get(CUST_FIELD_CODE_ID):
-        logging.warning("ShotGrid Missing Ayon ID")
+        # if the entity does not have an Ayon ID, try to create it
+        logging.warning("ShotGrid Missing Ayon ID.")
+
+        logging.debug(f"Creating Ayon Entity: {sg_ay_dict}")
+        try:
+            create_ay_entity_from_sg_event(
+                sg_event,
+                sg_project,
+                sg_session,
+                ayon_entity_hub,
+                sg_enabled_entities,
+                project_code_field,
+                custom_attribs_map
+            )
+        except Exception as e:
+            logging.error(f"Ayon Entity could not be created: {e}")
+            logging.warning(
+                "Skipping update of Ayon Entity. ShotGrid ID missing "
+                "and entity could not be created."
+            )
+        return
```

Comment:
Should the `return` be here??

Reply:
yeah I have added comment to explain it better